ros_ws/src/crazyswarm/scripts/pycrazyswarm/crazyflieSim.py

```diff
@@ -44,10 +48,15 @@ def step(self, duration):
     # should be called "animate" or something
     # but called "sleep" for source-compatibility with real-robot scripts
     def sleep(self, duration):
-        for t in np.arange(self.t, self.t + duration, self.dt):
-            self.visualizer.update(t, self.crazyflies)
+        # operator // has unexpected (wrong ?) behavior for this calculation.
+        ticks = math.floor((duration + self.sleepResidual) / self.dt)
+        self.sleepResidual += duration - self.dt * ticks
+        assert 0.0 <= self.sleepResidual < self.dt
+
+        for _ in range(int(ticks)):
+            self.visualizer.update(self.t, self.crazyflies)
             if self.output:
-                self.output.update(t, self.crazyflies)
+                self.output.update(self.t, self.crazyflies)
```

Comment:
shouldn't this be t rather than self.t?

Reply:
before, `t` was the iterate of `np.linspace`, but it would always be equal to `self.t` at the beginning of the loop because `self.step()` increments `self.t`. Now we are not using the `linspace` call so we use `self.t` instead.